baseapp/options.go

```diff
@@ -17,7 +19,7 @@ import (
 // for options that need access to non-exported fields of the BaseApp
 
 // SetPruning sets a pruning option on the multistore associated with the app
-func SetPruning(opts sdk.PruningOptions) func(*BaseApp) {
+func SetPruning(opts *pruningtypes.PruningOptions) func(*BaseApp) {
```

Comment:
Why the change?

Reply:
To prevent copying 42 bytes on every call in the worst case

Reply:
Is this function called more than once per run of the program? Even if it were called once per second, do you have a benchmark that shows a _meaningful_ performance improvement from this change? (By _meaningful_ I mean impactful to the overall performance of the application, rather than a microbenchmark showing an O(nanoseconds) improvement.)

Performance optimizations should be motivated by well-defined need, and justified by reproducible data.

Reply:
Only once. You're right, not very meaningful. I changed this back to not being a pointer.

Made a similar change for snapshot options (which I defined in this PR) for consistency.

Reply:
> > we should not call `Commit()` concurrently with `DeliverTx()` on baseapp. However, we rely on tendermint that this never happens at the same time
> 
> Does Tendermint actually provide that guarantee? And not in the "happy path" but always?

According to [the spec](https://docs.tendermint.com/master/introduction/what-is-tendermint.html#intro-to-abci), that's correct

Also, checked the code:

https://github.com/osmosis-labs/tendermint/blob/167fa738a379147f333b86f539bc03f597b978e5/state/execution.go#L161-L201

`DeliverTx(...)` happens in `execBlockOnProxyApp(...) `  while `Commit(...)` is on line 201.